src/exoticatechnologies/modifications/exotics/ExoticSpecialItemPlugin.kt

```diff
@@ -38,53 +37,89 @@ open class ExoticSpecialItemPlugin : ModSpecialItemPlugin() {
-                exoticData = ExoticData(modId!!)
-                if (exoticData!!.key != modId) {
-                    modId = exoticData!!.key
-                    stack.specialDataIfSpecial.data.replace(param, modId!!)
+//                exoticData = ExoticData(modId!!)
+//                if (exoticData!!.key != modId) {
+//                    modId = exoticData!!.key
+//                    stack.specialDataIfSpecial.data.replace(param, modId!!)
+//                }
+                modId?.let {safeModId ->
+                    exoticData = ExoticData(safeModId)
+                    exoticData?.let {safeExoticData ->
+                        if (safeExoticData.key != safeModId) {
+                            modId = safeExoticData.key
+                            stack.specialDataIfSpecial.data.replace(param, safeModId)
+                        }
+                    }
                 }
             }
 
             Param.EXOTIC_TYPE -> {
                 if (param == "true" || param == "false") {
                     val split = stack.specialDataIfSpecial.data
-                        .split(",".toRegex())
-                        .dropLastWhile { it.isEmpty() }
-                        .toTypedArray()
+                            .split(",".toRegex())
+                            .dropLastWhile { it.isEmpty() }
+                            .toTypedArray()
                     val newData = String.format("%s,NORMAL,%s", split[0], split[1])
                     stack.specialDataIfSpecial.data = newData //fix saves
                     ignoreCrate = param.toBoolean()
 
-                    exoticData!!.type = ExoticType.NORMAL
+//                    exoticData!!.type = ExoticType.NORMAL
+                    exoticData?.let {
+                        it.type = ExoticType.NORMAL
+                    }
                 } else {
-                    exoticData!!.type = ExoticType.valueOf(param)
+//                    exoticData!!.type = ExoticType.valueOf(param)
+                    exoticData?.let {
+                        it.type = ExoticType.valueOf(param)
+                    }
                 }
             }
 
             Param.IGNORE_CRATE -> ignoreCrate = param.toBoolean()
         }
     }
 
+    public fun getBasePrice(): Int {
+        // This is using an elvis operator because even when it's nullguarded properly
+        // kotlin complains that since 'exotic' is a mutable property, it could have changed
+        // and it just doesn't make any sense to use a 'let' call in a null-guarded part of the if()
+        //
+        // Of course I would have used 'return if(exotic != null) { return exotic.getBasePrice() } else { 250000 }
+        val price: Int = if (exotic != null) {
+            exotic!!.getBasePrice()
+        } else {
+//            log.error("exotic was null\t\texoticData: ${exoticData}, exoticData.exotic.name: ${exoticData?.exotic?.name}, exoticData.exotic.cost: ${exoticData?.exotic?.getBasePrice()}")
```

Comment:
no longer necessary, can always readd if/when it needs to be readded. remove please

Reply:
done

Reply:
resolving